File: app/src/main/java/nl/jpelgrm/movienotifier/data/DaoNotifications.java
package nl.jpelgrm.movienotifier.data;

import java.util.List;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import nl.jpelgrm.movienotifier.models.Notification;
@Dao
public interface DaoNotifications {
    @Insert
    void add(Notification notification);

    @Query("SELECT * FROM Notifications WHERE UserID = :userid ORDER BY time DESC")
    DataSource.Factory<Integer, Notification> getNotificationsForUser(String userid);

    @Query("SELECT * FROM Notifications WHERE WatcherID = :watcherid")
    List<Notification> getNotificationsForWatcher(String watcherid);

    @Update
    void update(Notification notification);

    @Delete
    void delete(Notification notification);
}

File: app/src/main/java/nl/jpelgrm/movienotifier/data/DaoNotifications.kt
package nl.jpelgrm.movienotifier.data

import androidx.paging.DataSource
import androidx.room.*
import nl.jpelgrm.movienotifier.models.Notification

@Dao
interface DaoNotifications {
    @Insert
    fun add(notification: Notification?)

    @Query("SELECT * FROM Notifications WHERE UserID = :userid ORDER BY time DESC")
    fun getNotificationsForUser(userid: String): DataSource.Factory<Int, Notification>

    @Query("SELECT * FROM Notifications WHERE WatcherID = :watcherid")
    fun getNotificationsForWatcher(watcherid: String): List<Notification>

    @Update
    fun update(notification: Notification)

    @Delete
    fun delete(notification: Notification)
}